### ISSUE TITLE
Crashes at boot with "Unexpected interrupt on CPU 0"

### ISSUE THREAD

**Issue by rwmjones** [memtest86plus/memtest86plus#180]
I compiled memtest86 @ commit 66b138934891599ba1e20383ada789358009a63d using:
```
$ cd build64
$ make
```
I copied the `memtest.efi` binary to `/boot` and booted it using:
```
# interrupt grub boot to get grub command line by pressing 'c'
grub> linux /memtest.efi
grub> boot
```
It bluescreens immediately with the attached error (note the screen really is cut off at the left hand side by the monitor I'm using).

I also tried setting `enable_trace = true` in `app/config.c` but the output was identical.

![IMG_20221007_121542](https://user-images.githubusercontent.com/259688/194541668-3e996850-e7cf-4e55-950e-80fb7b002c86.png)

**Comment by martinwhitaker** [comment]
Is this a recent regression, or is this the first time you've tried it?

**Comment by martinwhitaker** [comment]
Ah, looking at the screenshot, your XHCI controller is mapped above the 4GB boundary. Please could you test the fix I've just pushed (commit aaa0cff).

**Comment by rwmjones** [comment]
Yes that worked perfectly, and it's now doing the memtest, thanks!
